omisego-sdk/src/main/java/co/omisego/omisego/model/transaction/request/TransactionRequest.kt

```diff
@@ -7,17 +7,157 @@ package co.omisego.omisego.model.transaction.request
+
+    /**
+     * The minted token for the request
+     * In the case of a type "send", this will be the token taken from the requester
+     * In the case of a type "receive" this will be the token received by the requester
+     */
     val mintedToken: MintedToken,
-    val status: String,
-    val amount: Double?,
-    val address: String?
+
+    /**
+     * The amount of minted token to use for the transaction (down to subunit to unit)
+     * This amount needs to be either specified by the requester or the consumer
+     */
+    val amount: BigDecimal?,
+
+    /**
+     * The address from which to send or receive the minted tokens
+     */
+    val address: String?,
+
+    /**
+     * The user that initiated the request
+     */
+    val user: User?,
+
+    /**
+     * The topic which can be listened in order to receive events regarding this request
+     */
+    val socketTopic: String,
+
+    /**
+     * The maximum number of time that this request can be consumed
+     */
+    val maxConsumption: Int?,
+
+    /**
+     * The status of the request (valid or expired)
+     */
+    val status: TransactionRequestStatus,
+
+    /**
+     * Allow or not the consumer to override the amount specified in the request
+     */
+    val allowAmountOverride: Boolean,
+
+    /**
+     * A boolean indicating if the request needs a confirmation from the requester before being proceeded
+     */
+    val requireConfirmation: Boolean,
+
+    /**
+     * The date when the request will expire and not be consumable anymore
+     */
+    val expirationDate: Date,
+
+    /**
+     * The reason why the request expired
+     */
+    val expirationReason: String?,
+
+    /**
+     * The amount of time in millisecond during which a consumption is valid
+     */
+    val consumptionLifetime: Int?,
+
+    /**
+     * The creation date of the request
+     */
+    val createdAt: Date?,
+
+    /**
+     * The date when the request expired
+     */
+    val expiredAt: Date?
 )
+
+/**
+ * An extension function that converts the [TransactionRequest] to the [TransactionConsumptionParams] easily
+ *
+ * @param amount The amount of minted token to transfer (down to subunit to unit)
+ * @param address The address to use for the consumption
+ * @param tokenId The id of the minted token to use for the request
+ * In the case of a type "send", this will be the token that the consumer will receive
+ * In the case of a type "receive" this will be the token that the consumer will send
+ * @param idempotencyToken The idempotency token to use for the consumption
+ * @param correlationId An id that can uniquely identify a transaction. Typically an order id from a provider.
+ * @param metadata Additional metadata for the consumption
+ * @param encryptedMetadata Additional encrypted metadata for the consumption
+ *
+ * @return The [TransactionConsumptionParams] used for consume the a transaction request
+ */
+fun TransactionRequest.toTransactionConsumptionParams(
```

Comment:
If this file starts to get too big, moving this in a `converter` package could be a good idea. Not an issue for now 😄

Reply:
Good idea 👍